### PR TITLE
Follow-up of #400: REPL, Serialization, Distiller, Test of literals

### PR DESCRIPTION
**Pull request by imkiva**
This is part of #400

- [x] Better pretty printer
- [x] Serialization of literals
- [x] Rewrite more visitors using pattern matching.

### CODE REVIEW

**Comment by imkiva** [comment]
image

Change this to `main` when #400 is merged

File: base/src/test/resources/success/src/Literal.aya [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  | suc n => even n
+
+def even Nat : Bool
+  | 0 => true
+  | 1 => odd 0
+  | suc n => odd n
+
+def overlap largeInt Nat : Nat
+  | 0 => 0
+-- TODO: PatClassifier
```
Comment (ice1000) [inline base/src/test/resources/success/src/Literal.aya:69]: What's this `TODO`?

Comment (imkiva) [reply base/src/test/resources/success/src/Literal.aya:69]: It's about converting constructor to lit and vice versa in pat classifier so classifying large numbers won't crash (Stack overflow)

File: base/src/main/java/org/aya/distill/CoreDistiller.java [hunk 2]
```diff
@@ -151,7 +152,9 @@ yield visitCalls(false, term(Outer.AppHead, head), args.view(), outer,
       }
       case CallTerm.Struct structCall -> visitArgsCalls(structCall.ref(), STRUCT_CALL, structCall.args(), outer);
       case CallTerm.Data dataCall -> visitArgsCalls(dataCall.ref(), DATA_CALL, dataCall.args(), outer);
-      case LitTerm.ShapedInt shaped -> shaped.with(
+      case LitTerm.ShapedInt shaped -> options.map.get(DistillerOptions.Key.ShowLiterals)
+        ? Doc.plain(String.valueOf(shaped.repr()))
+        : shaped.with(
         (zero, suc) -> shaped.repr() == 0
           ? linkLit(0, zero.ref, CON_CALL)
           : linkLit(shaped.repr(), suc.ref, CON_CALL),
```
Comment (imkiva) [inline base/src/main/java/org/aya/distill/CoreDistiller.java:160]: Why change this? The literals won't be linked to its constructor when `ShowLiteral` is set.

Comment (imkiva) [reply base/src/main/java/org/aya/distill/CoreDistiller.java:160]: Like this 
image

**Comment by ice1000** [comment]
404!

**Pull request merged**